js/ui/workspace.js

```diff
@@ -1556,6 +1556,10 @@ const Workspace = new Lang.Class({
 
     _windowRemoved : function(metaWorkspace, metaWin) {
         this._doRemoveWindow(metaWin);
+
+        // Make sure to show the icon grid if we removed the last window
+        if (!Main.workspaceMonitor.hasActiveWindows)
+            Main.overview.showApps();
```

Comment:
I am afraid that we'll then have two code paths trying to show the desktop at the same time when the last window closes. I am OK deferring this to a separate ticket to be investigated later though.